ScoutSuite/providers/aws/resources/resources.py

```diff
@@ -0,0 +1,49 @@
+
+"""This module provides implementations for Resources and CompositeResources for AWS."""
+
+from ScoutSuite.providers.base.configs.resources import Resources, CompositeResources
+from ScoutSuite.providers.aws.facade.facade import AWSFacade
+import abc
+
+
+class AWSResources(Resources, metaclass=abc.ABCMeta):
+
+    """This is the base class for AWS resources."""
+
+    def __init__(self, scope: dict):
+        """
+        :param scope: The scope holds the scope in which the resource is located. This usually means \
+                      at least a region, but can also contain a VPC id, an owner id, etc. It should be \
+                      used when fetching the data through the facade.
+        """
+        
+        self.scope = scope
+        self.facade = AWSFacade()
+
+
+class AWSCompositeResources(AWSResources, CompositeResources, metaclass=abc.ABCMeta):
+
+    """This class represents a collection of AWSResources. Classes extending AWSCompositeResources should \
+    define a "_children" attribute which consists of a list of tuples describing the children. The tuples \
+    are expected to respect the following format: (<child_class>, <child_name>). The child_name is used by \
+    indicates the name under which the child will be stored in the parent object.
```

Comment:
*is used to indicate

Reply:
Would it be worth to mention than `None` can be passed as `<child_name>` to indicate that the content of the child will be directly included in the parent (as result of the dict operation `update()`)?

Reply:
(Please disregard my previous comment if you're actually not doing this with AWS... Just thought about it because I'm doing so with Azure.)

Reply:
Not sure this is a use case for AWS. Thanks for pointing it out tho!